x-pack/plugin/ml-feature-index-builder/src/test/java/org/elasticsearch/xpack/ml/featureindexbuilder/job/AggregationConfigTests.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.ml.featureindexbuilder.job;
+
+import org.apache.lucene.util.LuceneTestCase.AwaitsFix;
+import org.elasticsearch.common.io.stream.Writeable.Reader;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.search.aggregations.AggregationBuilder;
+import org.elasticsearch.search.aggregations.AggregationBuilders;
+import org.elasticsearch.search.aggregations.AggregatorFactories;
+import java.io.IOException;
+
+import static org.hamcrest.Matchers.equalTo;
+
+// broken upstream
+@AwaitsFix(bugUrl="https://github.com/elastic/elasticsearch/issues/33942")
+public class AggregationConfigTests extends AbstractSerializingFeatureIndexBuilderTestCase<AggregationConfig> {
+
+    public static AggregationConfig randonAggregationConfig() {
+        AggregatorFactories.Builder builder = new AggregatorFactories.Builder();
+
+        for (int i = 1; i < randomIntBetween(1, 20); ++i) {
```

Comment:
Is it valid for the number of aggregator factories to be 0?  If not maybe the `AggregationConfig` class itself should validate that it has at least 1?

Reply:
good catch, should be `for (int i = 0 ...`

Nevertheless, `AggregationConfig` misses validation, will be part of an upcoming PR.